main.cpp

```diff
@@ -0,0 +1,53 @@
+//
+// Created by Nikolay Fedotenko on 03.03.2021.
+//
+
+#include "CPU_consumption/Consumption/ConsumptionManager.h"
+
+/*! \brief
+ *  Main function, inits and runs everything
+ */
+int main()
+{
+    glfwInit();
+
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+    glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
+
+    glfwWindowHint(GLFW_RESIZABLE, GL_FALSE);
+
+    GLFWwindow* window = glfwCreateWindow(WINDOW_WIDTH, WINDOW_HEIGHT,
+                                          "CPU consumption monitor", nullptr, nullptr);
+
+    if (window == nullptr)
+    {
+        glfwTerminate();
+
+        return EXIT_FAILURE;
+    }
+
+    glfwMakeContextCurrent(window);
+
+    glewExperimental = GL_TRUE;
+
+    if (glewInit() != GLEW_OK)
+        return EXIT_FAILURE;
+
+    std::unique_ptr<ConsumptionManager> manager( new ConsumptionManager() );
+    manager->init();
```

Comment:
![image](https://user-images.githubusercontent.com/33322723/115275980-288ed300-a14b-11eb-9c74-ad62f45713b0.png)

Reply:
RAII is cool, this is **not** RAII

Reply:
> RAII is cool, this is **not** RAII

Why not? I consider smart pointers as the implementation of RAII. Do you think that the usage of smart pointers here is extra? If yes, would you prefer to use the C-style pointers in this case or just to create an object on stack?

Reply:
Ok, you have a smart pointer and create an object with `new` word, that's perfectly fine and means that once you're getting out of the scope a _destructor_ of your object will be called and the memory will be freed. 
BUT
1) There's no destructor for this object. And object contains some fields acquiring memory by `new`. So the memory won't be released correctly.
2) RAII for C++ implies presence of both _constructor_ allocating memory and _destructor_ releasing it. You're having separate method doing job of constructor. That's old-fashioned-Symbian-style (one of my previous comments points at it), that's not RAII at all.